nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java

```diff
@@ -177,10 +177,11 @@ public List<WorkflowExecutor> getExecutors() {
         });
   }
 
-  public void markShutdown() {
+  public void markShutdown(boolean graceful) {
     try {
-      jdbc.update("update nflow_executor set expires=current_timestamp, stopped=current_timestamp, recovered=null where "
-          + getExecutorGroupCondition() + " and id = ?", getExecutorId());
+      var sql = graceful ? "set expires=current_timestamp, stopped=current_timestamp, recovered=current_timestamp" :
+              "set expires=" + sqlVariants.currentTimePlusSeconds(timeoutSeconds) + ", stopped=current_timestamp, recovered=null";
+      jdbc.update("update nflow_executor " + sql + " where id = ?", getExecutorId());
```

Comment:
```suggestion
      var sql = "update nflow_executor set expires=current_timestamp, stopped=current_timestamp, recovered=? where id = ?";
      jdbc.update(sql, graceful ? "current_timestamp" : null, getExecutorId());
```

Reply:
I would really prefer to give some extra delay to the in-progress workflows when a node is not shut down gracefully, which means that the workflow threads are still executing. Just to reduce the likelihood that the same instance is running on two executors concurrently.

Reply:
Yes, the comment was outdated.